src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,38 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter formatter =
+            DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder builder = new StringBuilder();
+        try {
+            LocalDate dateStart = LocalDate.parse(dateFrom, formatter);
+            LocalDate dateFinish = LocalDate.parse(dateTo, formatter);
+            int salary = 0;
+            builder.append("Report for period ").append(dateFrom).append(" - ").append(dateTo);
+
+            for (String name : names) {
+                for (String str : data) {
+                    String[] values = str.split(" ");
+                    LocalDate dateWork = LocalDate.parse(values[0], formatter);
+                    if (dateWork.isAfter(dateStart.minusDays(1))
+                            && dateWork.isBefore(dateFinish.plusDays(1))
+                            && name.equals(values[1])) {
+                        salary += Integer.parseInt(values[2]) * Integer.parseInt(values[3]);
```

Comment:
Same here, it's not obvious what is stored under values[2] and values[2] so it's better give extract and give proper names